fix: makes defineIntegration option defaults optional (#29)
florian-lefebvre/astro-integration-kit: docs/src/content/docs/utilities/define-integration.mdx

The `defaults` configuration accepts the default values for your integration's options.
`defaults` will be deeply merged with the options provided by the user. Please note that
this is not runtime safe, it's good practice to check the types of your options inside `setup`.

:::caution

> When writing your options types, be sure to be very explicit about them, specifically concerning
> optional properties. Make sure to specify `undefined` as a possible value:

Author: ```suggestion
When typing optional properties within your `IntegrationOptions`, make sure to specify `undefined` as a possible value.
```

```ts "undefined"
interface IntegrationOptions {
    foo?: string | undefined;
}
```

> We recommend that you extend the strictest astro tsconfig, that enforces such practise:

Author: ```suggestion
We recommend that you extend the strictest Astro TSConfig. This will help you create a more stable integration across many platforms and environments.
```
